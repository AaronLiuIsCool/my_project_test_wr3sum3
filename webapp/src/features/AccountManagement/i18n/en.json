{
    "heading": "账号管理",
    "basicInfo": "基本信息",
    "accountManagement": "账号管理",
    "wechat": "微信",
    "email": "邮箱",
    "password": "密码",
    "bind": "前往绑定",
    "unbind": "解除绑定"
  }