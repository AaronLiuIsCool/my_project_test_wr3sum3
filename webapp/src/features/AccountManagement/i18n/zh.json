{
  "heading": "账号管理",
  "basicInfo": "基本信息",
  "accountManagement": "账号管理",
  "wechat": "微信",
  "email": "邮箱",
  "password": "密码",
  "bind": "前往绑定",
  "unbind": "解除绑定",
  "binded": "已绑定",
  "notBinded": "未绑定",
  "reset": "重置",
  "enterEmailAddress": "请输入您所要绑定的邮箱",
  "save": "保存",
  "avatar": "头像",
  "name": "姓名",
  "updateEmail": "更换邮箱",
  "gender": "性别",
  "identity": "身份",
  "genderOptions": [
    {
      "key": "male",
      "value": "男"
    },
    {
      "key": "female",
      "value": "女"
    }
  ],
  "identityOptions": [
    {
      "key": "male",
      "value": "男"
    },
    {
      "key": "female",
      "value": "女"
    }
  ]
}
