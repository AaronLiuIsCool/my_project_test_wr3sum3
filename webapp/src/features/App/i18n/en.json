{
  "brand": "Tailoryourvc",

  "nav_item_resumehub": "Resume Hub",
  "nav_item_jobmatcher": "Jobs",

  "language_toggle_zh": "中文",
  "language_toggle_en": "English",

  "logout": "Sign out",
  "defaultInitial": "Unknown"
}