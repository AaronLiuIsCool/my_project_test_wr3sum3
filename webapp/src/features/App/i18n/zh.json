{
  "brand": "快刀简历",

  "nav_item_resumehub": "我的简历",
  "nav_item_jobmatcher": "工作搜索",
  "nav_item_jobcollction": "我的岗位",
  "language_toggle_zh": "中文",
  "language_toggle_en": "English",
  "createResume": "创建简历",
  "createNewResume": "创建一份全新简历",
  "chooseLanguage": "请选择简历语言",
  "resumeUpload": "我有简历了，我要上传",
  "resumeUploadSupportType": "支持PDF/Word 文档，文件最大支持20MB",
  "uploadYourResume": "从本地上传",
  "upload": "上传",
  "logout": "登出",
  "defaultInitial": "未知",
  "name_it":"请您为这份新简历命名",
  "name_suggestion":"建议用简历类别，比如 [咨询] [投行] [战投] 或者是 [岗位名称+公司名称] 的形式来来命名以方便查询",
  "confirm":"确认",
  "RPreview": {
    "linkedinLink": "领英链接",
    "githubLink": "Github链接",
    "educationBackground": "教育背景",
    "workExperience": "工作经历",
    "projectExperience": "项目经历",
    "studentWorkAndVolunteer": "志愿服务",
    "certificateAndAward": "技能证书 & 获得荣誉",
    "awards": "获得荣誉",
    "certificate": "技能证书",
    "validForever": "永久有效",
    "expiredAt": "到期",
    "current": "现今",
    "editThemeColor": "编辑主题色",
    "smartTranslation": "智能翻译",
    "translationEN": "English",
    "translationCN": "中文版式",
    "oneClickWholePage": "一键整页",
    "downloadResume": "下载简历",
    "perparingResume": "生成中..",
    "downLoadResume": "下载简历",
    "linkedIn": "领英链接",
    "weblink": "项目集链接"
  }
}
