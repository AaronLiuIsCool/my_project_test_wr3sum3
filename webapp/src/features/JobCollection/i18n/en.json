{
    "heading": "我的岗位",
    "all": "全部",
    "applied": "已经申请",
    "postedDate": "发布日期",
    "jobLocation": "岗位所在地",
    "applyJob": "申请工作"
  }