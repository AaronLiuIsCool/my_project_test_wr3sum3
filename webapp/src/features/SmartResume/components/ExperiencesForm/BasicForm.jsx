import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { Row, Col, Form } from 'react-bootstrap';

import InputGroup from 'components/InputGroup';
import DropdownGroup from 'components/DropdownGroup';
import Button from 'react-bootstrap/Button';

import { actions, selectId } from '../../slicer';
import AvatarUpload from './AvatarUpload';

import { validateBasic, validateBasicEntry } from '../../slicer/basic';
import { updateStatus, updateAllStatus } from '../../slicer/common';
import { adaptBasics } from '../../utils/servicesAdaptor';
import { dispatchUpdates, updateRating, updateCityOptions } from '../../utils/resume';
import ResumeServices from 'shell/services/ResumeServices';
import { getLogger } from 'shell/logger';

import { previewResume } from '../ResumePreview/resumeBuilder';

// json data for dropdowns
import countryOptions from 'data/country.json';
import { Summary } from '../Summary';
import { useEffect } from 'react';
import ArrowUp from '../../assets/arrow-up.svg'; 

const logger = getLogger('BasicForm');
const resumeServices = new ResumeServices();
const fields = [
    'nameCn',
    'email',
    'phone',
    'city',
    'linkedin',
    'weblink',
    'country'
];

const BasicForm = ({ data, photoReference, completed, messages }) => {
	const [showSummary, setShowSummary] = useState(false);
	const resumeId = useSelector(selectId);
  const [validated, setValidated] = useState(false);
  const [cityOptions, setCityOptions] = useState([]);
	const [status, setStatus] = useState({
		nameCn: {},
		email: {},
		phone: {},
		city: {},
		linkedin: {},
    weblink: {},
    country: {}
	});
	useEffect(() => {
    setShowSummary(Boolean(completed));
  }, [completed])
  useEffect(() => {
    updateCityOptions(data.country, setCityOptions);
  }, [data.country]);
	const dispatch = useDispatch();
  const toggleShowSummary = () => {
    setShowSummary(!showSummary);
  };
	const save = async () => {
		previewResume(messages.RPreview);
		let id = data.id;
		try {
			const responseJson =  data.id === undefined ? 
				await resumeServices.createBasics(resumeId, adaptBasics({ completed: true, data })) :
				await resumeServices.updateBasics(resumeId, adaptBasics({ completed: true, data }));
			id = id || responseJson.id;
		} catch (exception) {
			logger.error(exception);
		} finally {
			dispatch(actions.updateBasicsId({ id }));
			dispatchUpdates('update-score');
		}
	};

	const handleSubmit = async (event) => {
    event.preventDefault();
    event.stopPropagation();
    updateAllStatus(validateBasicEntry, status, setStatus, fields, data);
    if (!validateBasic(data)) {
      console.log(111111)
      setValidated(false);
      return;
    }
    toggleShowSummary();
    setValidated(true);
    dispatch(actions.completeBasic());
    
    await save();
    updateRating();
  };

	const handleNameCnChange = (event) => {
		const value = event.target.value;
		updateStatus(validateBasicEntry, status, setStatus, 'nameCn', value);
		dispatch(actions.updateNameCn({ value }));
	};

	const handleEmailChange = (event) => {
		const value = event.target.value;
		updateStatus(validateBasicEntry, status, setStatus, 'email', value);
		dispatch(actions.updateEmail({ value }));
	};

	const handlePhoneChange = (event) => {
		const value = event.target.value;
		updateStatus(validateBasicEntry, status, setStatus, 'phone', value);
		dispatch(actions.updatePhone({ value }));
  };
  
  const handleCountryChange = (values) => {
    const value = values.length === 0 ? null : values[0].data;
    updateCityOptions(value, setCityOptions);
    updateStatus(validateBasicEntry, status, setStatus, 'country', value);
    dispatch(actions.updateCountry({ value }));
  }

	const handleCityChange = (values) => {
		const value = values.length === 0 ? null : values[0].data;
		if (!validateBasicEntry('city', value)) {
			return;
		}
		updateStatus(validateBasicEntry, status, setStatus, 'city', value);
		dispatch(actions.updateCity({ value }));
	};

	const handleLinkedinChange = (event) => {
		const value = event.target.value;
		updateStatus(validateBasicEntry, status, setStatus, 'linkedin', value);

		dispatch(actions.updateLinkedin({ value }));
	};

	const handleWeblinkChange = (event) => {
		const value = event.target.value;
		updateStatus(validateBasicEntry, status, setStatus, 'weblink', value);
		dispatch(actions.updateWeblink({ value }));
	};
	return (
    <div className="form_body">
      {showSummary ? (
        <div>
          <Summary
            name={data.nameCn}
            avatar={photoReference?.url}
            type={'BasicInfo'}
            handleClickCallback={toggleShowSummary}
          />
        </div>
      ) : (
        <Form validated={validated} onSubmit={handleSubmit}>
          <Row className="flexie">
            <AvatarUpload photoReference={photoReference} />
            <div className="toggle-up-arrow" onClick={toggleShowSummary}>
              {completed && <img src={ArrowUp} alt="up-arrow"/>}
            </div>
          </Row>
          <Row>
            <Col>
              <InputGroup
                label={messages.cnName}
                id="basic-name-cn"
                placeholder={messages.enterCnName}
                value={data.nameCn}
                onChange={handleNameCnChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.nameCn.isValid}
                isInvalid={status.nameCn.isInvalid}
              />
            </Col>
            <Col lg="6">
              <InputGroup
                label={messages.email}
                id="basic-email"
                placeholder={messages.enterEmail}
                value={data.email}
                onChange={handleEmailChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.email.isValid}
                isInvalid={status.email.isInvalid}
              />
            </Col>
          </Row>

          <Row>
            
            <Col lg="6">
              <InputGroup
                label={messages.phone}
                id="basic-phone"
                placeholder={messages.enterPhone}
                value={data.phone}
                onChange={handlePhoneChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.phone.isValid}
                isInvalid={status.phone.isInvalid}
              />
            </Col>
            <Col lg="3">
              <DropdownGroup
                label={messages.country}
                id="basic-country"
                placeholder={messages.country}
                options={countryOptions}
                value={data.country}
                onChange={handleCountryChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.country.isValid}
                isInvalid={status.country.isInvalid}
              />
            </Col>
            <Col lg="3">
              <DropdownGroup
                label={messages.schoolCity}
                id="education-city"
                placeholder={messages.schoolCity}
                options={cityOptions}
                value={data.city}
                onChange={handleCityChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.city.isValid}
                isInvalid={status.city.isInvalid}
              />
            </Col>
          </Row>
          <Row>
            <Col>
              <InputGroup
                label={messages.linkedin}
                id="basic-linkedin"
                placeholder={messages.enterLinkedin}
                value={data.linkedin}
                onChange={handleLinkedinChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.linkedin.isValid}
                isInvalid={status.linkedin.isInvalid}
              />
            </Col>
            <Col>
              <InputGroup
                label={messages.weblink}
                id="basic-weblink"
                placeholder={messages.enterWeblink}
                value={data.weblink}
                onChange={handleWeblinkChange}
                feedbackMessage={messages.entryIsInvalid}
                isValid={status.weblink.isValid}
                isInvalid={status.weblink.isInvalid}
              />
            </Col>
          </Row>
          {/* todo: reset button  */}
          <Row className="form_buttons">
            <Col className="space_betweens">
              {/* just a placeholder so we do need to change the css */}
              <p className="hidden"></p>
              <Button variant="primary" type="submit">
                {messages.save}
              </Button>
            </Col>
          </Row>
        </Form>
      )}
    </div>
  );
};

BasicForm.propTypes = {
	data: PropTypes.object.isRequired,
	completed: PropTypes.bool.isRequired,
	messages: PropTypes.object.isRequired,
};

export default BasicForm;
