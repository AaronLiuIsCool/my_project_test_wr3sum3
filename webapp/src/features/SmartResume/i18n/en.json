{
  "title": "Smart Resume",

  "basicInfo": "Basic information",
  "education": "Education background",
  "workXp": "Work experiences",
  "projectXp": "Project experiences",
  "otherXp": "Other experiences",
  "certifications": "Skills/Certifications"
}