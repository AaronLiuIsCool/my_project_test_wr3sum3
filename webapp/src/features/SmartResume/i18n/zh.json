{
  "title": "智能创建",
  "basicInfo": "基本信息",
  "enterBasicInfo": "请输入您的一些基本个人信息",
  "education": "教育背景",
  "educationInfo": "请添加关于您教育背景的相关信息",
  "workXp": "工作经历",
  "projectXp": "项目经历",
  "otherXp": "志愿者经历",
  "certifications": "技能证书",
  "writeAssistant": "经历写作助手",
  "addNewEducationExperience": "添加教育背景",
  "addNewCertificate": "添加证书",
  "addNewExperience": "添加新经历",
  "uploadPhoto": "上传照片",
  "addPhoto": "添加头像",
  "samples": "什么样的照片试用于简历?",
  "cnName": "中文姓名",
  "enterCnName": "请输入您的中文全名",
  "enName": "英文名称",
  "enterEnName": "First & Last Name",
  "email": "邮箱",
  "enterEmail": "请填写您的邮箱地址",
  "phone": "联系电话",
  "enterPhone": "输入您的联系电话",
  "city": "城市",
  "enterCity": "您现在的居住城市",
  "linkedin": "个人领英 Linkedin 链接",
  "enterLinkedin": "请填写您的Linkedin链接",
  "weblink": "个人 Github 或者作品集链接",
  "enterWeblink": "请填写您的Github链接或作品集链接",
  "resetForm": "清空重填",
  "confirm": "确认",
  "enterSchoolInfo": "填写院校信息",
  "schoolName": "院校名称",
  "enterSchoolName": "请输入学校名字",
  "graduateGPA": "毕业GPA",
  "enterSchoolDate": "入学日期",
  "graduateDate": "毕业日期",
  "major": "专业",
  "enterMajor": "输入所在的专业",
  "degree": "学历",
  "enterDegree": "请输入您获得的学历",
  "schoolCity": "所在城市",
  "schoolCountry": "学校所在国家",
  "highestAward": "所获最高奖项 (有任何奖项都是加分好事！)",
  "enterHighestAward": "请填写您在校期间所获得的最高奖项",
  "otherAward": "其他获得奖项",
  "enterOtherAward": "请填写您在校期间所获得的其他奖项",
  "save": "保存",
  "entryIsInvalid": "请输入正确的信息",
  "enterNewExperience": "填写新经历",
  "enterNewCertificate": "填写新证书",
  "work": "工作或实习经历",
  "workInfo": "请您尽可能的表述您前10份的工作/实习/项目经历。从您最近经历开始说起。",
  "yes": "是",
  "no": "否",
  "workName": "岗位名称",
  "enterWorkName": "请填写您的岗位名称",
  "stillAtWork": "是否已离职？",
  "companyName": "单位/公司名称",
  "enterCompanyName": "请填写您所在的单位/公司名称",
  "workStartDate": "入职日期",
  "workEndDate": "离职日期",
  "workCity": "岗位所在城市",
  "country": "国家",
  "workCountry": "岗位所在国家",
  "workDetailsDescription": "岗位细节描述",
  "enterWorkDetailsDescription": "说说您在职的时所做的事情",
  "project": "项目经历",
  "projectInfo": "请您尽可能的表述您前10份的工作/实习/项目经历。从您最近经历开始说起。",
  "participateRole": "项目名称",
  "enterParticipateRole": "请填写所参与项目的名称",
  "endOrNot": "是否已结束？",
  "projectStartDate": "项目开始时间",
  "projectEndDate": "项目结束时间",
  "projectCity": "项目所在城市",
  "projectCountry": "项目所在国家",
  "projectDetailsDescription": "项目细节描述",
  "enterProjectDetailsDescription": "参与身份",
  "volunteerExperience": "志愿者经历",
  "enterVolunteerExperience": "请您尽可能的表述您志愿者经历。从您最近经历开始说起。",
  "leaveOrNot": "是否已离开？",
  "institutionName": "机构名称",
  "enterInstitutionName": "请填报名参与志愿者的机构的名称",
  "volunteerStartDate": "志愿经历开始时间",
  "volunteerEndDate": "志愿经历结束时间",
  "experienceCity": "经历所在城市",
  "experienceCountry": "经历所在国家",
  "volunteerDetailsDescription": "志愿者经历细节描述",
  "enterVolunteerDetailsDescription": "说说您在职的时所做的事情",
  "certificate": "技能证书",
  "certificateInfo": "请添加关于您教育背景的相关信息",
  "certificateName": "证书名称",
  "enterCertificateName": "请输入证书的名称",
  "validForever": "永久有效？",
  "issueDate": "颁发时间/获得时间",
  "expireDate": "有效期至",
  "yymmdd": "年/月",
  "dateDisplayFormat": "YYYY/MM",
  "clickForDetails": "点击查看强度详情",
  "score_A": "优秀",
  "score_B": "良好",
  "score_C": "及格",
  "score_D": "薄弱",
  "score_A_MSG": "简历强度:简历信息完整, 表达专业 牛人你真棒！",
  "score_B_MSG": "简历强度:简历信息完整, 参考‘写作助手’专业术语继续提升吧！",
  "score_C_MSG": "简历强度:及格, 简历信息有所缺失, 继续加油哦！",
  "score_D_MSG": "简历强度:薄弱, 简历信息较少, 请前往补全更多信息！",
  "RPreview": {
    "linkedinLink": "领英链接",
    "githubLink": "Github链接",
    "educationBackground": "教育背景",
    "workExperience": "工作经历",
    "projectExperience": "项目经历",
    "studentWorkAndVolunteer": "志愿服务",
    "certificateAndAward": "技能证书 & 获得荣誉",
    "awards": "获得荣誉",
    "certificate": "技能证书",
    "validForever": "永久有效",
    "expiredAt": "到期",
    "current": "现今",
    "editThemeColor": "编辑主题色",
    "smartTranslation": "智能翻译",
    "oneClickWholePage": "一键整页",
    "downloadResume": "下载简历",
    "perparingResume": "生成中..",
    "downLoadResume": "下载简历",
    "linkedIn": "领英链接",
    "weblink": "项目集链接"
  },
  "resumeTips": {
    "resumeTips": "简历编写小帖士"
  },
  "photoMessage": "如果您投递中国地区的岗位，建议您添加个人简历照片",
  "personalWebsite":"个人网站",
  "websiteMessage": "填写个人网站会显得更专业哟！",
  "linkedinAccount": "领英账号",
  "linkedinMessage": "填写领英账号会显得更专业哟！",
  "resume": "简历",
  "notAlign":"页数不工整",
  "layoutMessage":"您的简历，尚未进行整页排版，请使用<b>“一键整页”</b>让我们帮您自动优化排版！",
  "highestAwards": "所获最高奖项",
  "highestAwardsMessage": "填写所获最高奖项显得更专业哟！",
  "otherAwardMessage": "填写其他获得奖项显得更专业哟！",
  "gpaMessageOne": "您的GPA在3.5以上，建议您填写，当作一个亮点来展示",
  "gpaMessageTwo": "您的GPA，在3.0以下，不建议您添加到简历上",
  "gpaMessageThree": "您的GPA在3-3.5之间，您可以填写在简历上呢",
  "schoolRating1": "是世界名校，非常棒哟！",
  "schoolRating2": "知名度蛮高的，可以可以！",
  "schoolRating3": "是很有特色的一所院校，还不错哟！",
  "schoolPopularity": "学校知名度",
  "good": "挺好",
  "verygood": "很好",
  "extremegood": "非常好",
  "certificationsMessage": "技能证书，是非常重要的资质证明，您这块背景较为薄弱，建议您查看快课，快速掌握专业技能，提升背景竞争力",
  "workXpLengthOverTwo": "您已经提供3+段工作经历，数量充实哟，建议您查看经历写作助手，提高表达专业性",
  "workXpLengthTwo": "您已经提供2段工作经历，数量上，还是可以的，建议您查看经历写作助手，提高表达专业性",
  "workXpLengthOne": "您目前只有1段工作经历，显得比较单薄，如果可以的话，建议您再添加一段相关经历，提高竞争力。",
  "workXpLengthZero": "您目前没有工作经历，是一个明显软肋，如果可以的话，建议您再添加一段相关经历，提高竞争力。",
  
  "projectXpLengthOverTwo": "您已经提供3+段专业项目经历，数量充实哟，建议您查看经历写作助手，提高表达专业性",
  "projectXpLengthTwo": "您已经提供2段专业项目经历，数量上，还是可以的，建议您查看经历写作助手，提高表达专业性",
  "projectXpLengthOne": "您目前只有1段专业项目经历，显得比较单薄，如果可以的话，建议您再添加一段相关经历，提高竞争力。",
  "projectXpLengthZero": "您目前没有专业项目经历，如果可以的话，建议您再添加一段相关经历，提高竞争力。",
  
  "otherXpLengthOverTwo": "您已经提供3+段志愿者经历，数量充实哟，建议您查看经历写作助手，提高表达专业性",
  "otherXpLengthTwo": "志愿者经历，可以更加全面展示，专业技能之外的特质、能力，比如人际交往、活动策划等等，您目前只有2段志愿者经历，可以参考“经历写作助手”再多补充一些",
  "otherXpLengthOne": "志愿者经历，可以更加全面展示，专业技能之外的特质、能力，比如人际交往、活动策划等等，您目前只有1段志愿者经历，可以参考“经历写作助手”再多补充一些",
  "otherXpLengthZero": "志愿者经历，可以更加全面展示，专业技能之外的特质、能力，比如人际交往、活动策划等等，您目前没有志愿者经历，可以参考“经历写作助手”再多补充一些",
  
  "workExp": "工作实习经历",
  "projectExp": "项目经历",
  "otherExp": "志愿者经历",
  "sorted": "时间排序",
  "sortedMessage": "您的经历模块，还未按时间排序，“一键排序”帮您自动排序",
  "employedAt1": "您曾经就职的单位",
  "employedAt2": "，是行业",
  "employedAt3": "龙头之一",
  "employedAt4": "明星企业",
  "employedAt5": "，这段经历含金量很棒哟！",
  "nationalInfluence": "就业于全球知名单位！",
  "globalInfluence": "就业于全国知名单位！",
  "graduatedFrom": "您的毕业院校{school}，",
  "ratingDetail": "简历强度分析详情",
  "notEnter": "您没有填写",
  "jumpTo": "前往",
  "expVal": "经历含金量",
  "tooLess": "数量过少",
  "entireLayout": "简历整体版式",
  "all": "全部",
  "expLackOf": "经历详情缺乏",
  "keywords": "专业术语",
  "keywordsMessage": "您以下的{exp}经历详情描述没有用任何行业术语。请参考”写作助手”里面的写作建议进行修改！",
  "expression": "表达专业度",
  "exp": "经历",
  "notQuantify": "成就表述未被量化",
  "quantifyMessage": "您以下的{exp}经历详情描述没有用任何量化关键词。请参考”写作助手”里面的写作建议进行修改！",
  "hasProblem": "有问题",
  "sort": "排序",
  "workTitle": "工作/实习经历",
  "projTitle": "项目经历",
  "volTitle": "志愿者经历",
  "infoComplete": "信息完整度",
  "layout": "简历版式",
  "expValue": "经历含金量",
  "present": "至今",
  "valid_through": "有效期至: ",
  "forever": "永久有效",
  "delete": "删除",
  "basicInfoNotCompleted": "您还未填写基本信息",
  "EduInfoNotCompleted": "您还未填写教育背景",
  "expReplacement": {
    "workXp": "工作实习",
    "projectXp": "项目",
    "otherXp": "志愿者"
  },
  "datePicker": {
    "lang": {
      "placeholder": "请选择日期",
      "yearPlaceholder": "请选择年份",
      "quarterPlaceholder": "请选择季度",
      "monthPlaceholder": "请选择月份",
      "weekPlaceholder": "请选择周",
      "rangePlaceholder": [
        "开始日期",
        "结束日期"
      ],
      "rangeYearPlaceholder": [
        "开始年份",
        "结束年份"
      ],
      "rangeMonthPlaceholder": [
        "开始月份",
        "结束月份"
      ],
      "rangeWeekPlaceholder": [
        "开始周",
        "结束周"
      ],
      "locale": "zh_CN",
      "today": "今天",
      "now": "此刻",
      "backToToday": "返回今天",
      "ok": "确 定",
      "timeSelect": "选择时间",
      "dateSelect": "选择日期",
      "weekSelect": "选择周",
      "clear": "清除",
      "month": "月",
      "year": "年",
      "previousMonth": "上个月 (翻页上键)",
      "nextMonth": "下个月 (翻页下键)",
      "monthSelect": "选择月份",
      "yearSelect": "选择年份",
      "decadeSelect": "选择年代",
      "yearFormat": "YYYY年",
      "dayFormat": "D日",
      "dateFormat": "YYYY年M月D日",
      "dateTimeFormat": "YYYY年M月D日 HH时mm分ss秒",
      "previousYear": "上一年 (Control键加左方向键)",
      "nextYear": "下一年 (Control键加右方向键)",
      "previousDecade": "上一年代",
      "nextDecade": "下一年代",
      "previousCentury": "上一世纪",
      "nextCentury": "下一世纪"
    },
    "timePickerLocale": {
      "placeholder": "请选择时间",
      "rangePlaceholder": [
        "开始时间",
        "结束时间"
      ]
    }
  },
  "quantifyMetMessage": "您以下经历描述，对经历过程、结果，进行了量化表达，可以可以！！介绍过往经历，量化过程以及成就，展示了结果导向的表述思路，是现实工作场景中非常倡导的，您还可以参考“经历撰写助手”，进一步完善经历描述：",
  "quantifyMetHeader": "经历详情已量化表达！",
  "keywordsMetHeader": "经历详情已使用专业术语！",
  "keywordsMetMessage": "您以下经历描述，使用到了专业术语，非常棒！！HR或者ATS，会通过捕捉专业术语，来判断您的背景资质，与岗位要求，是否吻合，您还可以参考“经历撰写助手”，进一步提炼经历描述：",
  "keyword": "关键字："
}