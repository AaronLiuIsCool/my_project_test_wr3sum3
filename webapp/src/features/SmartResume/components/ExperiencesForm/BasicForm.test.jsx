import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { I8nContext } from 'shell/i18n';
import renderer from 'react-test-renderer';

import 'react-dates/initialize';

import store from 'store';
import BasicForm from './BasicForm';

jest.mock('./AvatarUpload', () => () => <div>Avatar</div>)

const BasicFormWithProvider = (props) => (
  <Provider store={store}>
    <I8nContext.Provider value={{}}>
      <BasicForm {...props} />
    </I8nContext.Provider>
  </Provider>
);

describe('BasicForm form', () => {
  test('should render successfully with input', () => {
    const wrapper = mount(
      <BasicFormWithProvider data={{}} messages={{}} index={0} />
    );
    expect(wrapper.find('Form').length).toEqual(1);
  });

  test('It should render <Summary />', () => {
    const data = {
      id: 'test',
    };
    const wrapper = mount(
      <BasicFormWithProvider
        data={data}
        messages={{}}
        index={0}
        completed={true}
      />
    );
    expect(wrapper.find('Summary').length).toEqual(1);
  });

  test('It should render <Summary /> after click', () => {
    const data = {};
    const wrapper = mount(
      <BasicFormWithProvider data={data} messages={{}} index={0} />
    );
    const btn = wrapper.find('.toggle-up-arrow');
    btn.simulate('click');
    expect(wrapper.find('Summary').length).toEqual(1);
  });

  test('It should match snapshot', () => {
    const data = {
      nameCn: '中文',
      nameEn: 'English',
      email: '123@asd.ca',
      phone: '1231231321',
      city: '北京',
      linkedin: 'http://www.google.ca',
      weblink: 'http://www.google.ca',
    };
    const tree = renderer
      .create(<BasicFormWithProvider data={data} messages={{}} index={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
