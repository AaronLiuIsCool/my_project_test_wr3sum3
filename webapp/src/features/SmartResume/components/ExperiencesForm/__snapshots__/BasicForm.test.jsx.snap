// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicForm form It should match snapshot 1`] = `
<div
  className="form_body"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="flexie row"
    >
      <div>
        Avatar
      </div>
      <div
        className="toggle-up-arrow"
        onClick={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="basic-name-cn"
          />
          <input
            className="form-control"
            id="basic-name-cn"
            onChange={[Function]}
            type="text"
            value="中文"
          />
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="basic-name-en"
          />
          <input
            className="form-control"
            id="basic-name-en"
            onChange={[Function]}
            type="text"
            value="English"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-6"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="basic-email"
          />
          <input
            className="form-control"
            id="basic-email"
            onChange={[Function]}
            type="text"
            value="123@asd.ca"
          />
        </div>
      </div>
      <div
        className="col-lg-3"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="basic-phone"
          />
          <input
            className="form-control"
            id="basic-phone"
            onChange={[Function]}
            type="text"
            value="1231231321"
          />
        </div>
      </div>
      <div
        className="col-lg-3"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="education-city"
          />
          <div
            className="rbt"
            style={
              Object {
                "outline": "none",
                "position": "relative",
              }
            }
            tabIndex={-1}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "height": "100%",
                  "position": "relative",
                }
              }
            >
              <input
                aria-autocomplete="both"
                aria-expanded={false}
                aria-haspopup="listbox"
                autoComplete="off"
                className="rbt-input-main form-control rbt-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="combobox"
                type="text"
                value="北京"
              />
              <input
                aria-hidden={true}
                className="rbt-input-hint"
                readOnly={true}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "transparent",
                    "boxShadow": "none",
                    "color": "rgba(0, 0, 0, 0.35)",
                    "left": 0,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
                tabIndex={-1}
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="basic-linkedin"
          />
          <input
            className="form-control"
            id="basic-linkedin"
            onChange={[Function]}
            type="text"
            value="http://www.google.ca"
          />
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="form_item form-validation-wrapper form-group"
        >
          <label
            className="form-label"
            htmlFor="basic-weblink"
          />
          <input
            className="form-control"
            id="basic-weblink"
            onChange={[Function]}
            type="text"
            value="http://www.google.ca"
          />
        </div>
      </div>
    </div>
    <div
      className="form_buttons row"
    >
      <div
        className="space_betweens col"
      >
        <p
          className="hidden"
        />
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        />
      </div>
    </div>
  </form>
</div>
`;
